pages/live/live.js

```diff
@@ -123,15 +126,44 @@ function receiveMessage (event) {
         document.querySelector('#realtimeEnabledText').textContent =  'Current Call Disabled'
       break
     case MESSAGE_TYPES.APP_TO_EXTERNAL_END_CALL:
+      if ('sessionId' in message && message['sessionId'] !== null) {
+        // If we have explicit session ids and they don't match, don't stop anything yet.
+        // This keeps us online if an old call's end comes after a new call starts.
+        //
+        // Since we don't have a session id at the time of START_CALL,
+        // this could get us stuck (until the next call begins and ends) with
+        // _call_active = true and the coaching toggle disabled if we get
+        // the END_CALL before the START_COACHING (startSession).
+        // But at least we guarantee that no session remains stuck open
+        // as long as we ever receive its END_CALL:
+        // 1. If END_CALL is received for a past session, it will already have
+        //    been ended by startSession calling endSession;
+        // 2. If END_CALL is received for a current session, call endSession
+        //    when COACHING_END comes down the coaching websocket,
+        //    and set a timeout in endCurrentConversation() as a backup.
+        // 3. If END_CALL is received for a future session, record it in
+        //    _sessionIdsEnded so we can reject its start when that comes.
+        //
```

Comment:
so it seems like this is basically as good as can be done. in general the best you can do is disconnect at min(`END_CALL`, some other session's `START_CALL`), and i think this does that.  if we loose an END_CALL message, we have no easy way of knowing the call is over or not so we can't do anything. 

if we wanted to, it would probably be easy to send `callId`s at both START_CALL and END_CALL, but it's not clear to me this buys us anything